apps/openchallenges/infra/src/preview-instance-alb/preview-instance-alb.ts

```diff
@@ -54,10 +54,12 @@ export class PreviewInstanceAlb extends Construct {
       tags: { Name: `${nameTagPrefix}-preview-instance-tg` },
     });
 
-    this.listener = new LbListener(this, 'preview_instance_alb_http_80', {
+    this.listener = new LbListener(this, 'preview_instance_alb_https_443', {
       loadBalancerArn: this.lb.arn,
-      port: 80,
-      protocol: 'HTTP',
+      port: 443,
```

Comment:
Could redirect http to https at ALB

Reply:
Good idea!

Reply:
Entering `http://dev.openchallenges.io` in the browser actually already redirects to `https://dev.openchallenges.io`.

When using curl, there is no response for `http://`:

```console
$ curl http://dev.openchallenges.io
curl: (7) Failed to connect to dev.openchallenges.io port 80 after 8 ms: Connection refused
```

We get a response when using `https://`:

```console
$ curl https://dev.openchallenges.io
<!DOCTYPE html><html lang="en"><head><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="">
    <meta charset="utf-8">
    <title>OpenChallenges</title>
    <base href="/">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="icon" type="image/x-icon" href="favicon.ico">
...
```

Reply:
https://docs.bridgecrew.io/docs/ensure-that-alb-redirects-http-requests-into-https-ones

Reply:
I have added a listener to redirect port 80 to 443. Now the response from `curl` looks like:

```console
$ curl http://dev.openchallenges.io
<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
</body>
</html>
```

And when telling `curl` to follow the redirection:

```console
$ curl -L http://dev.openchallenges.io
<!DOCTYPE html><html lang="en"><head><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="">
    <meta charset="utf-8">
    <title>OpenChallenges</title>
    <base href="/">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="icon" type="image/x-icon" href="favicon.ico">
...
```

Reply:
Reference: https://docs.bridgecrew.io/docs/ensure-that-alb-redirects-http-requests-into-https-ones